common/set_benchmark.cpp

```diff
@@ -0,0 +1,366 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <benchmark/benchmark.h>
+
+#include "absl/container/flat_hash_set.h"
+#include "common/raw_hashtable_benchmark_helpers.h"
+#include "common/set.h"
+#include "llvm/ADT/DenseSet.h"
+
+namespace Carbon {
+namespace {
+
+using RawHashtable::CarbonHashDI;
+using RawHashtable::GetKeysAndHitKeys;
+using RawHashtable::GetKeysAndMissKeys;
+using RawHashtable::HitArgs;
+using RawHashtable::SizeArgs;
+using RawHashtable::ValueToBool;
+
+template <typename SetT>
+struct IsCarbonSetImpl : std::false_type {};
+template <typename KT, int MinSmallSize>
+struct IsCarbonSetImpl<Set<KT, MinSmallSize>> : std::true_type {};
+
+template <typename MapT>
+static constexpr bool IsCarbonSet = IsCarbonSetImpl<MapT>::value;
+
+template <typename SetT>
+struct SetWrapperImpl {
```

Comment:
Please comment how this is a wrapper to allow benchmarking Carbon and non-Carbon set implementations using a single API.